PutAndVerify/WriteAndVerify.tla

```diff
@@ -0,0 +1,65 @@
+--------------------------- MODULE WriteAndVerify ---------------------------
+
+CONSTANT CLIENTS
+
+VARIABLE storage, clState
+
+WVTypeOk == /\ storage \in [ CLIENTS -> { "none", "empty", "written" } ]
+            /\ clState \in [ CLIENTS -> { "begin", "precheck-done", "blank-written", 
+                                          "write-check-done", "committed", "aborted" } ]
+
+WVInit == /\ storage = [ cli \in CLIENTS |-> "none" ]
+          /\ clState = [ cli \in CLIENTS |-> "begin" ]
+
+(* Only one execution can be committed. *)
+(* Once committed, the file has been written. *)
+WVConsistency == /\ \A i, j \in CLIENTS : 
+                    (* If i != j, clState[i] and clState[j] cannot be both committed. *)
+                    i /= j /\ clState[i] = "committed" => ~clState[j] = "committed"
+                 /\ \A i, j \in CLIENTS : 
+                    i /= j /\ storage[i] = "written" => ~storage[j] = "written"
+                 /\ \A i \in CLIENTS : clState[i] = "committed" <=> storage[i] = "written"
+
+(* "Verify" condition: storage only contains our file or nothing. *)
+Check(c) == /\ \A f \in CLIENTS : f = c \/ storage[f] = "none"
+
+Precheck(c) ==    /\ clState[c] = "begin" 
+                  /\ IF Check(c) 
+                     THEN clState' = [ clState EXCEPT ![c] = "precheck-done" ]
+                     ELSE clState' = [ clState EXCEPT ![c] = "aborted" ]
+                  /\ UNCHANGED << storage >>
+
+WriteBlank(c) == /\ clState[c] = "precheck-done"
+                 /\ storage' = [ storage EXCEPT ![c] = "empty" ]
+                 /\ clState' = [ clState EXCEPT ![c] = "blank-written" ]
+
+ConflictCheck(c) == /\ clState[c] = "blank-written" 
+                    /\ IF Check(c) 
+                       THEN clState' = [ clState EXCEPT ![c] = "write-check-done" ]
+                       ELSE clState' = [ clState EXCEPT ![c] = "aborted" ]
+                    /\ UNCHANGED << storage >>
+                    
+(* Delete the file when aborted. *)
+RollBack(c) == /\ clState[c] = "aborted"
+               /\ storage[c] /= "none"

```

Comment:
is this necessary?

Reply:
In fact not. This won't create new states but just adding arrows between states. However our implmentation includes this, so I left it here.